From f07ff3efa3f04d56720766ee063e72be989b211e Mon Sep 17 00:00:00 2001
From: itsmanjeet <itsmanjeet1998@gmail.com>
Date: Fri, 28 May 2021 18:55:26 +0530
Subject: [PATCH] Fixed to search rlxos system images

---
 util/grub.d/10_linux.in | 27 +++++++++++++++++----------
 1 file changed, 17 insertions(+), 10 deletions(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 4532266be..0748c716c 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -164,7 +164,7 @@ machine=`uname -m`
 case "x$machine" in
     xi?86 | xx86_64)
 	list=
-	for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-* ; do
+	for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-* /boot/vmlinuz; do
 	    if grub_file_is_not_garbage "$i" ; then list="$list $i" ; fi
 	done ;;
     *) 
@@ -192,9 +192,14 @@ title_correction_code=
 submenu_indentation=""
 
 is_top_level=true
+
+savedlist="$list"
+for rlxos_system_image in $(ls /run/initramfs/rlxos/system | sort -r) ; do
+gettext_printf "System Image: %s\n" "$rlxos_system_image" >&2
+list="$savedlist"
 while [ "x$list" != "x" ] ; do
   linux=`version_find_latest $list`
-  gettext_printf "Found linux image: %s\n" "$linux" >&2
+  gettext_printf "  with kernel: %s\n" "$linux" >&2
   basename=`basename $linux`
   dirname=`dirname $linux`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
@@ -218,7 +223,8 @@ while [ "x$list" != "x" ] ; do
 	   "initramfs-genkernel-${version}" \
 	   "initramfs-genkernel-${alt_version}" \
 	   "initramfs-genkernel-${GENKERNEL_ARCH}-${version}" \
-	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}"; do
+	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}" \
+     "initrd"; do
     if test -e "${dirname}/${i}" ; then
       initrd_real="${i}"
       break
@@ -233,7 +239,7 @@ while [ "x$list" != "x" ] ; do
     for i in ${initrd}; do
       initrd_display="${initrd_display} ${dirname}/${i}"
     done
-    gettext_printf "Found initrd image: %s\n" "$(echo $initrd_display)" >&2
+    gettext_printf "  with initrd: %s\n" "$(echo $initrd_display)" >&2
   fi
 
   config=
@@ -262,8 +268,8 @@ while [ "x$list" != "x" ] ; do
   fi
 
   if [ "x$is_top_level" = xtrue ] && [ "x${GRUB_DISABLE_SUBMENU}" != xy ]; then
-    linux_entry "${OS}" "${version}" simple \
-    "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+    linux_entry "${OS}-${rlxos_system_image}" "${version}" simple \
+    "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} system=${rlxos_system_image}"
 
     submenu_indentation="$grub_tab"
     
@@ -275,15 +281,16 @@ while [ "x$list" != "x" ] ; do
     is_top_level=false
   fi
 
-  linux_entry "${OS}" "${version}" advanced \
-              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+  linux_entry "${OS}-${rlxos_system_image}" "${version}" advanced \
+              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} system=${rlxos_system_image}"
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
-    linux_entry "${OS}" "${version}" recovery \
-                "single ${GRUB_CMDLINE_LINUX}"
+    linux_entry "${OS}-${rlxos_system_image}" "${version}" recovery \
+                "single ${GRUB_CMDLINE_LINUX} system=${rlxos_system_image}"
   fi
 
   list=`echo $list | tr ' ' '\n' | fgrep -vx "$linux" | tr '\n' ' '`
 done
+done
 
 # If at least one kernel was found, then we need to
 # add a closing '}' for the submenu command.
-- 
2.28.0

