--- org/util/grub.d/10_linux.in	2020-12-23 04:54:17.000000000 +0530
+++ grub-2.06/util/grub.d/10_linux.in	2021-08-30 14:41:02.441971669 +0530
@@ -164,7 +164,7 @@
 case "x$machine" in
     xi?86 | xx86_64)
 	list=
-	for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-* ; do
+	for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-* /boot/vmlinuz ; do
 	    if grub_file_is_not_garbage "$i" ; then list="$list $i" ; fi
 	done ;;
     *) 
@@ -192,9 +192,13 @@
 submenu_indentation=""
 
 is_top_level=true
+saved_list="${list}"
+for rlxos_sys_image in $(ls /run/initramfs/rlxos/system | sort -r) ; do
+    gettext_printf "System Image: %s\n" "${rlxos_sys_image}" >&2
+    list="${saved_list}"
 while [ "x$list" != "x" ] ; do
   linux=`version_find_latest $list`
-  gettext_printf "Found linux image: %s\n" "$linux" >&2
+  gettext_printf "   linux image: %s\n" "$linux" >&2
   basename=`basename $linux`
   dirname=`dirname $linux`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
@@ -218,7 +222,8 @@
 	   "initramfs-genkernel-${version}" \
 	   "initramfs-genkernel-${alt_version}" \
 	   "initramfs-genkernel-${GENKERNEL_ARCH}-${version}" \
-	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}"; do
+	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}" \
+     "initrd"; do
     if test -e "${dirname}/${i}" ; then
       initrd_real="${i}"
       break
@@ -233,7 +238,7 @@
     for i in ${initrd}; do
       initrd_display="${initrd_display} ${dirname}/${i}"
     done
-    gettext_printf "Found initrd image: %s\n" "$(echo $initrd_display)" >&2
+    gettext_printf "   initrd image: %s\n" "$(echo $initrd_display)" >&2
   fi
 
   config=
@@ -271,7 +276,7 @@
 
   if [ "x$is_top_level" = xtrue ] && [ "x${GRUB_DISABLE_SUBMENU}" != xtrue ]; then
     linux_entry "${OS}" "${version}" simple \
-    "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+    "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} system=${rlxos_sys_image}"
 
     submenu_indentation="$grub_tab"
     
@@ -284,14 +289,15 @@
   fi
 
   linux_entry "${OS}" "${version}" advanced \
-              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} system=${rlxos_sys_image}"
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
     linux_entry "${OS}" "${version}" recovery \
-                "single ${GRUB_CMDLINE_LINUX}"
+                "single ${GRUB_CMDLINE_LINUX} system=${rlxos_sys_image}"
   fi
 
   list=`echo $list | tr ' ' '\n' | fgrep -vx "$linux" | tr '\n' ' '`
 done
+done
 
 # If at least one kernel was found, then we need to
 # add a closing '}' for the submenu command.
