From 899509dcfb41d692b73b39248ca99fdcdeaac371 Mon Sep 17 00:00:00 2001
From: "manjeet.singh1" <manjeet.singh@bobble.ai>
Date: Sat, 15 Jan 2022 13:17:45 +0530
Subject: [PATCH] updated path

---
 cmd/server/server.go | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/cmd/server/server.go b/cmd/server/server.go
index 687d7a0..ee297c6 100644
--- a/cmd/server/server.go
+++ b/cmd/server/server.go
@@ -3,6 +3,8 @@ package main
 import (
 	"errors"
 	"log"
+	"os"
+	"path"
 
 	"github.com/godbus/dbus/v5"
 	"github.com/godbus/dbus/v5/introspect"
@@ -39,7 +41,11 @@ func (b boltServer) Speak(query string) *dbus.Error {
 }
 
 func main() {
-	bot, err := bolt.Create("database")
+	localdatabase := path.Join(os.Getenv("HOME"), ".config", "bolt", "database")
+	if _, err := os.Stat(localdatabase); err != nil {
+		os.MkdirAll(localdatabase, 0755)
+	}
+	bot, err := bolt.Create(localdatabase)
 	server := boltServer{
 		backend: bot,
 	}
-- 
2.32.0

