From f8388002dccfae78bc77d4e4b9439b0b74740f13 Mon Sep 17 00:00:00 2001
From: itsmanjeet <itsmanjeet1998@gmail.com>
Date: Wed, 2 Aug 2023 11:03:58 +0200
Subject: [PATCH] do-not-override-system-theme

---
 src/adw-style-manager.c | 36 +++++++++++++++++++++++-------------
 1 file changed, 23 insertions(+), 13 deletions(-)

diff --git a/src/adw-style-manager.c b/src/adw-style-manager.c
index 42733169..6f77b3eb 100644
--- a/src/adw-style-manager.c
+++ b/src/adw-style-manager.c
@@ -229,6 +229,22 @@ notify_high_contrast_cb (AdwStyleManager *self)
   g_object_notify_by_pspec (G_OBJECT (self), props[PROP_HIGH_CONTRAST]);
 }
 
+static void
+adw_style_manager_theme_changed(GSettings *settings, const char *key, GdkDisplay *display)
+{
+  char *theme = g_settings_get_string(settings, "gtk-theme");
+  if (!theme)
+    {
+      theme = "Adwaita-empty";
+    }
+  gboolean is_dark_variant = NULL != strstr(theme, "-dark");
+
+  g_object_set (gtk_settings_get_for_display (display),
+                "gtk-theme-name", theme,
+                "gtk-application-prefer-dark-theme", is_dark_variant,
+                NULL);
+}
+
 static void
 adw_style_manager_constructed (GObject *object)
 {
@@ -254,19 +270,13 @@ adw_style_manager_constructed (GObject *object)
                              G_CONNECT_SWAPPED);
 
     if (!adw_is_granite_present () && !g_getenv ("GTK_THEME")) {
-      g_object_set (gtk_settings_get_for_display (self->display),
-                    "gtk-theme-name", "Adwaita-empty",
-                    NULL);
-
-      self->provider = gtk_css_provider_new ();
-      gtk_style_context_add_provider_for_display (self->display,
-                                                  GTK_STYLE_PROVIDER (self->provider),
-                                                  GTK_STYLE_PROVIDER_PRIORITY_THEME);
-
-      self->colors_provider = gtk_css_provider_new ();
-      gtk_style_context_add_provider_for_display (self->display,
-                                                  GTK_STYLE_PROVIDER (self->colors_provider),
-                                                  GTK_STYLE_PROVIDER_PRIORITY_THEME);
+      GSettingsSchemaSource *schema_source =
+          g_settings_schema_source_get_default();
+      GSettingsSchema *schema = g_settings_schema_source_lookup(
+          schema_source, "org.gnome.desktop.interface", true);
+      GSettings *interface_settings = g_settings_new_full(schema, NULL, NULL);
+      adw_style_manager_theme_changed(interface_settings, "gtk-theme", self->display);
+      g_signal_connect (interface_settings, "changed", G_CALLBACK (adw_style_manager_theme_changed), self->display);
     }
 
     self->animations_provider = gtk_css_provider_new ();
-- 
2.34.1

