--- old	2021-11-17 12:20:47.806154746 +0530
+++ new	2021-11-17 12:21:03.699210814 +0530
@@ -78,7 +78,7 @@
         names = g_new0 (gchar *, n_workspaces + 1);
         for(i = 0; i < n_workspaces; ++i) {
             if(G_LIKELY(i != workspace))
-                names[i] = g_strdup_printf(_("Workspace %d"), i + 1);
+                names[i] = g_strdup_printf(_("%d"), i + 1);
             else
                 names[i] = g_strdup(new_name);
         }
@@ -93,7 +93,7 @@
             names = g_realloc(names, sizeof(gchar *) * (n_workspaces + 1));
             for(i = prop_len; i < n_workspaces; ++i) {
                 if(i != workspace)
-                    names[i] = g_strdup_printf(_("Workspace %d"), i + 1);
+                    names[i] = g_strdup_printf(_("%d"), i + 1);
                 else
                     names[i] = g_strdup(new_name);
             }
