--- old	2021-11-16 14:29:34.586765843 +0530
+++ new	2021-11-16 14:33:40.653950409 +0530
@@ -31,7 +31,7 @@
 mousewheel_rollup=true
 move_opacity=100
 placement_mode=center
-placement_ratio=20
+placement_ratio=80
 popup_opacity=100
 prevent_focus_stealing=false
 raise_delay=250
@@ -48,18 +48,18 @@
 shadow_delta_y=-3
 shadow_opacity=50
 show_app_icon=false
-show_dock_shadow=true
+show_dock_shadow=false
 show_frame_shadow=true
-show_popup_shadow=false
+show_popup_shadow=true
 snap_resist=false
 snap_to_border=true
 snap_to_windows=false
 snap_width=10
 sync_to_vblank=false
-theme=Default
+theme=Dracula
 tile_on_move=true
 title_alignment=center
-title_font=Sans Bold 9
+title_font=Noto Sans Regular 10
 title_horizontal_offset=0
 titleless_maximize=false
 title_shadow_active=false
