--- 10_linux	2022-02-02 22:19:17.669635600 +0530
+++ 10_linux_new	2022-02-02 22:16:52.104527361 +0530
@@ -273,36 +273,31 @@
   if [ "x${GRUB_DISABLE_SUBMENU}" = xyes ] || [ "x${GRUB_DISABLE_SUBMENU}" = xy ]; then
     GRUB_DISABLE_SUBMENU="true"
   fi
+  
+  linux_entry "${OS} ${rlxos_sys_image}" "${version}" simple \
+  "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} system=${rlxos_sys_image}"
 
-  if [ "x$is_top_level" = xtrue ] && [ "x${GRUB_DISABLE_SUBMENU}" != xtrue ]; then
-    linux_entry "${OS}" "${version}" simple \
-    "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} system=${rlxos_sys_image}"
-
-    submenu_indentation="$grub_tab"
+  submenu_indentation="$grub_tab"
     
-    if [ -z "$boot_device_id" ]; then
-	boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
-    fi
-    # TRANSLATORS: %s is replaced with an OS name
-    echo "submenu '$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote)' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' {"
-    is_top_level=false
+  if [ -z "$boot_device_id" ]; then
+    boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
   fi
+  # TRANSLATORS: %s is replaced with an OS name
+  echo "submenu '$(gettext_printf "More options for %s" "${OS}" | grub_quote)' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' {"
 
-  linux_entry "${OS}" "${version}" advanced \
+  linux_entry "${OS} ${rlxos_sys_image}" "${version}" advanced \
               "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} system=${rlxos_sys_image}"
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
-    linux_entry "${OS}" "${version}" recovery \
+    linux_entry "${OS} ${rlxos_sys_image}" "${version}" recovery \
                 "single ${GRUB_CMDLINE_LINUX} system=${rlxos_sys_image}"
   fi
+  
+  echo '}'
+ 
 
   list=`echo $list | tr ' ' '\n' | fgrep -vx "$linux" | tr '\n' ' '`
 done
 done
 
-# If at least one kernel was found, then we need to
-# add a closing '}' for the submenu command.
-if [ x"$is_top_level" != xtrue ]; then
-  echo '}'
-fi
 
 echo "$title_correction_code"
