From 7b369b140998a7fe2a8f4465e597ca6ecc810a9b Mon Sep 17 00:00:00 2001
From: itsmanjeet <itsmanjeet1998@gmail.com>
Date: Wed, 10 Nov 2021 17:08:53 +0530
Subject: [PATCH] better looks  patch for xfce appfinder

---
 src/appfinder-window.c | 117 +++++++----------------------------------
 src/main.c             |  19 +++++--
 2 files changed, 34 insertions(+), 102 deletions(-)

diff --git a/src/appfinder-window.c b/src/appfinder-window.c
index 1f72bd3..0a67397 100644
--- a/src/appfinder-window.c
+++ b/src/appfinder-window.c
@@ -158,8 +158,6 @@ struct _XfceAppfinderWindow
 
   GdkPixbuf                  *icon_find;
 
-  GtkWidget                  *button_launch;
-
   GarconMenuDirectory        *filter_category;
   gchar                      *filter_text;
 
@@ -212,10 +210,10 @@ xfce_appfinder_window_init (XfceAppfinderWindow *window)
   GtkTreeViewColumn  *column;
   GtkCellRenderer    *renderer;
   GtkTreeSelection   *selection;
-  GtkWidget          *button;
   GtkEntryCompletion *completion;
   GtkCellRenderer    *cell;
   gint                integer;
+  GdkPixbuf          *pixbuf;
 
   window->channel = xfconf_channel_get ("xfce4-appfinder");
   window->last_window_height = xfconf_channel_get_int (window->channel, "/last/window-height", DEFAULT_WINDOW_HEIGHT);
@@ -228,6 +226,7 @@ xfce_appfinder_window_init (XfceAppfinderWindow *window)
   xfconf_g_property_bind (window->channel, "/item-icon-size", G_TYPE_UINT,
                           G_OBJECT (window->model), "icon-size");
 
+  gtk_window_set_decorated (GTK_WINDOW (window), FALSE);
   gtk_window_set_title (GTK_WINDOW (window), _("Application Finder"));
   integer = xfconf_channel_get_int (window->channel, "/last/window-width", DEFAULT_WINDOW_WIDTH);
   gtk_window_set_default_size (GTK_WINDOW (window), integer, -1);
@@ -238,12 +237,12 @@ xfce_appfinder_window_init (XfceAppfinderWindow *window)
       gtk_window_set_position (GTK_WINDOW (window), GTK_WIN_POS_CENTER);
     }
 
-  vbox = gtk_box_new (GTK_ORIENTATION_VERTICAL, 6);
+  vbox = gtk_box_new (GTK_ORIENTATION_VERTICAL, 0);
   gtk_container_add (GTK_CONTAINER (window), vbox);
-  gtk_container_set_border_width (GTK_CONTAINER (vbox), 6);
+  gtk_container_set_border_width (GTK_CONTAINER (vbox), 0);
   gtk_widget_show (vbox);
 
-  hbox = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 6);
+  hbox = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
   gtk_box_pack_start (GTK_BOX (vbox), hbox, FALSE, TRUE, 0);
   gtk_widget_show (hbox);
 
@@ -251,9 +250,9 @@ xfce_appfinder_window_init (XfceAppfinderWindow *window)
   window->image = image = gtk_image_new_from_pixbuf (window->icon_find);
   gtk_widget_set_size_request (image, 48, 48);
   gtk_widget_set_halign(image, GTK_ALIGN_CENTER);
-  gtk_container_add (GTK_CONTAINER (hbox), image);
-  gtk_widget_show (image);
-  
+  // gtk_container_add (GTK_CONTAINER (hbox), image);
+  // gtk_widget_show (image);
+
   window->entry = entry = gtk_entry_new ();
   gtk_widget_set_halign(entry, GTK_ALIGN_FILL);
   gtk_widget_set_valign (entry, GTK_ALIGN_CENTER);
@@ -271,18 +270,13 @@ xfce_appfinder_window_init (XfceAppfinderWindow *window)
       G_CALLBACK (xfce_appfinder_window_entry_key_press_event), window);
   gtk_entry_set_icon_from_icon_name (GTK_ENTRY (entry),
                                      GTK_ENTRY_ICON_SECONDARY,
-                                     XFCE_APPFINDER_STOCK_GO_DOWN);
+                                     "configure");
   gtk_entry_set_icon_tooltip_text (GTK_ENTRY (entry),
                                    GTK_ENTRY_ICON_SECONDARY,
                                    _("Toggle view mode"));
+
   gtk_widget_show (entry);
 
-  window->completion = completion = gtk_entry_completion_new ();
-  gtk_entry_completion_set_model (completion, GTK_TREE_MODEL (window->model));
-  gtk_entry_completion_set_match_func (completion, xfce_appfinder_window_completion_match_func, window, NULL);
-  g_object_set (G_OBJECT (completion), "text-column", XFCE_APPFINDER_MODEL_COLUMN_COMMAND, NULL);
-  gtk_entry_completion_set_popup_single_match (completion, TRUE);
-  gtk_entry_completion_set_inline_completion (completion, TRUE);
 
   cell = gtk_cell_renderer_text_new ();
   g_object_set (G_OBJECT (cell), "ellipsize", PANGO_ELLIPSIZE_END, NULL);
@@ -305,7 +299,7 @@ xfce_appfinder_window_init (XfceAppfinderWindow *window)
   gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (scroll), GTK_POLICY_NEVER, GTK_POLICY_AUTOMATIC);
   gtk_widget_show (scroll);
 
-  if (xfconf_channel_get_bool (window->channel, "/hide-category-pane", FALSE))
+  if (xfconf_channel_get_bool (window->channel, "/hide-category-pane", TRUE))
     gtk_widget_set_visible (scroll, FALSE);
 
   sidepane = window->sidepane = gtk_tree_view_new_with_model (GTK_TREE_MODEL (window->category_model));
@@ -349,32 +343,6 @@ xfce_appfinder_window_init (XfceAppfinderWindow *window)
   /* set the icon or tree view */
   xfce_appfinder_window_view (window);
 
-  hbox = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
-  gtk_box_pack_start (GTK_BOX (vbox), hbox, FALSE, TRUE, 0);
-  gtk_widget_show (hbox);
-
-  button = gtk_button_new_with_mnemonic (_("_Preferences"));
-  gtk_box_pack_start (GTK_BOX (hbox), button, FALSE, FALSE, 0);
-  g_signal_connect (G_OBJECT (button), "clicked",
-      G_CALLBACK (xfce_appfinder_window_preferences), window);
-  gtk_widget_set_valign (button, GTK_ALIGN_CENTER);
-  gtk_button_set_always_show_image(GTK_BUTTON(button), TRUE);
-  gtk_widget_show (button);
-
-  image = gtk_image_new_from_icon_name (XFCE_APPFINDER_STOCK_PREFERENCES, GTK_ICON_SIZE_BUTTON);
-  gtk_button_set_image (GTK_BUTTON (button), image);
-
-  window->button_launch = button = gtk_button_new_with_mnemonic (_("La_unch"));
-  gtk_box_pack_end (GTK_BOX (hbox), button, FALSE, FALSE, 0);
-  g_signal_connect_swapped (G_OBJECT (button), "clicked",
-      G_CALLBACK (xfce_appfinder_window_launch_clicked), window);
-  gtk_widget_set_sensitive (button, FALSE);
-  gtk_button_set_always_show_image(GTK_BUTTON(button), TRUE);
-  gtk_widget_show (button);
-
-  image = gtk_image_new_from_icon_name (XFCE_APPFINDER_STOCK_EXECUTE, GTK_ICON_SIZE_BUTTON);
-  gtk_button_set_image (GTK_BUTTON (button), image);
-
   window->icon_theme = gtk_icon_theme_get_for_screen (gtk_window_get_screen (GTK_WINDOW (window)));
   g_signal_connect_swapped (G_OBJECT (window->icon_theme), "changed",
       G_CALLBACK (xfce_appfinder_window_icon_theme_changed), window);
@@ -1280,8 +1248,6 @@ xfce_appfinder_window_entry_changed_idle (gpointer data)
     }
   else
     {
-      gtk_widget_set_sensitive (window->button_launch, IS_STRING (text));
-
       /* Clear entry's error icon and tooltip */
       gtk_entry_set_icon_from_icon_name (GTK_ENTRY (window->entry), GTK_ENTRY_ICON_PRIMARY, NULL);
       gtk_entry_set_icon_tooltip_text (GTK_ENTRY (window->entry), GTK_ENTRY_ICON_PRIMARY, NULL);
@@ -1344,9 +1310,9 @@ xfce_appfinder_window_entry_activate (GtkEditable         *entry,
       if (has_selection)
         xfce_appfinder_window_execute (window, TRUE);
     }
-  else if (gtk_widget_get_sensitive (window->button_launch))
+  else
     {
-      gtk_button_clicked (GTK_BUTTON (window->button_launch));
+      xfce_appfinder_window_execute (window, TRUE);
     }
 }
 
@@ -1376,6 +1342,8 @@ xfce_appfinder_window_entry_key_press_event (GtkWidget           *entry,
   gboolean          is_expanded;
   GtkTreePath      *path;
 
+  xfce_appfinder_window_set_expanded (window, TRUE);
+
   if (event->keyval == GDK_KEY_Tab &&
       !gtk_widget_get_visible (window->paned) &&
       xfce_appfinder_window_pointer_is_grabbed (entry))
@@ -1384,38 +1352,7 @@ xfce_appfinder_window_entry_key_press_event (GtkWidget           *entry,
       return TRUE;
     }
 
-  if (event->keyval == GDK_KEY_Up ||
-      event->keyval == GDK_KEY_Down)
-    {
-      expand = (event->keyval == GDK_KEY_Down);
-      is_expanded = gtk_widget_get_visible (window->paned);
-
-      if (is_expanded != expand)
-        {
-          /* don't break entry completion navigation in collapsed mode */
-          if (!is_expanded && xfce_appfinder_window_pointer_is_grabbed (entry))
-            {
-              /* window is still collapsed and the pointer is grabbed
-               * by the popup menu, do nothing with the event */
-              return FALSE;
-            }
-
-          xfce_appfinder_window_set_expanded (window, expand);
-          return TRUE;
-        }
-
-      /* The first item is usually selected, so we should jump to 2nd one
-       * when down key is pressed */
-      if (is_expanded && expand &&
-          GTK_IS_TREE_VIEW (window->view) && /* does not make sense for icon view */
-          xfce_appfinder_window_view_get_selected_path (window, &path))
-        {
-          gtk_tree_path_next (path);
-          gtk_tree_view_set_cursor (GTK_TREE_VIEW (window->view), path, NULL, FALSE);
-          gtk_tree_path_free (path);
-        }
-    }
-
+  
   return FALSE;
 }
 
@@ -1538,7 +1475,7 @@ xfce_appfinder_window_entry_icon_released (GtkEntry             *entry,
                                            XfceAppfinderWindow  *window)
 {
   if (icon_pos == GTK_ENTRY_ICON_SECONDARY)
-    xfce_appfinder_window_set_expanded (window, !gtk_widget_get_visible (window->paned));
+    xfce_appfinder_preferences_show (gtk_widget_get_screen (entry));
 }
 
 
@@ -1713,7 +1650,6 @@ xfce_appfinder_window_item_changed (XfceAppfinderWindow *window)
   if (gtk_widget_get_visible (window->paned))
     {
       can_launch = xfce_appfinder_window_view_get_selected (window, &model, &iter);
-      gtk_widget_set_sensitive (window->button_launch, can_launch);
 
       if (can_launch)
         {
@@ -1736,8 +1672,7 @@ xfce_appfinder_window_item_changed (XfceAppfinderWindow *window)
 static void
 xfce_appfinder_window_row_activated (XfceAppfinderWindow *window)
 {
-  if (gtk_widget_get_sensitive (window->button_launch))
-    gtk_button_clicked (GTK_BUTTON (window->button_launch));
+  xfce_appfinder_window_execute (window, TRUE);
 }
 
 
@@ -1844,12 +1779,6 @@ xfce_appfinder_window_execute (XfceAppfinderWindow *window,
   screen = gtk_window_get_screen (GTK_WINDOW (window));
   if (gtk_widget_get_visible (window->paned))
     {
-      if (!gtk_widget_get_sensitive (window->button_launch))
-        {
-          only_custom_cmd = TRUE;
-          goto entry_execute;
-        }
-
       if (xfce_appfinder_window_view_get_selected (window, &model, &iter))
         {
           child_model = model;
@@ -1876,10 +1805,7 @@ xfce_appfinder_window_execute (XfceAppfinderWindow *window,
     }
   else
     {
-      if (!gtk_widget_get_sensitive (window->button_launch))
-        return;
-
-      entry_execute:
+       entry_execute:
 
       text = gtk_entry_get_text (GTK_ENTRY (window->entry));
       save_cmd = TRUE;
@@ -1969,11 +1895,6 @@ xfce_appfinder_window_set_expanded (XfceAppfinderWindow *window,
   /* show/hide pane with treeviews */
   gtk_widget_set_visible (window->paned, expanded);
 
-  /* toggle icon */
-  gtk_entry_set_icon_from_icon_name (GTK_ENTRY (window->entry), GTK_ENTRY_ICON_SECONDARY,
-                                     expanded ? XFCE_APPFINDER_STOCK_GO_UP : XFCE_APPFINDER_STOCK_GO_DOWN);
-  gtk_entry_set_icon_from_icon_name (GTK_ENTRY (window->entry), GTK_ENTRY_ICON_PRIMARY, NULL);
-
   /* update completion (remove completed text of restart completion) */
   completion = gtk_entry_get_completion (GTK_ENTRY (window->entry));
   if (completion != NULL)
diff --git a/src/main.c b/src/main.c
index 85258b4..317aa91 100644
--- a/src/main.c
+++ b/src/main.c
@@ -43,7 +43,7 @@
 
 
 
-static gboolean            opt_collapsed = FALSE;
+static gboolean            opt_collapsed = TRUE;
 static gboolean            opt_version = FALSE;
 static gboolean            opt_replace = FALSE;
 static gboolean            opt_quit = FALSE;
@@ -209,9 +209,12 @@ appfinder_window_new (const gchar *startup_id,
 gint
 main (gint argc, gchar **argv)
 {
-  GError      *error = NULL;
-  const gchar *startup_id;
-  GSList      *windows_destroy;
+  GError         *error = NULL;
+  const gchar    *startup_id;
+  GSList         *windows_destroy;
+  GtkCssProvider *provider;
+  GdkDisplay     *display;
+  GdkScreen      *screen;
 
   /* set translation domain */
   xfce_textdomain (GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR, "UTF-8");
@@ -301,6 +304,14 @@ main (gint argc, gchar **argv)
   appfinder_refcount_debug_init ();
 #endif
 
+  provider = gtk_css_provider_new ();
+  gtk_css_provider_load_from_data (GTK_CSS_PROVIDER (provider),
+    "entry { border: none; padding: 12px 20px; }", -1, NULL);
+
+  display = gdk_display_get_default ();
+  screen  = gdk_display_get_default_screen (display);
+  gtk_style_context_add_provider_for_screen (screen, GTK_STYLE_PROVIDER (provider), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+
   /* create initial window */
   appfinder_window_new (NULL, !opt_collapsed);
 
-- 
2.32.0

