From 4987e58bc21910623d0f6d8f959d35a6e4f6297f Mon Sep 17 00:00:00 2001
From: itsmanjeet <itsmanjeet1998@gmail.com>
Date: Mon, 11 Apr 2022 10:53:59 +0530
Subject: [PATCH] added rlxos logo

---
 neofetch | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/neofetch b/neofetch
index 48b96d21..324ccabf 100755
--- a/neofetch
+++ b/neofetch
@@ -10237,6 +10237,29 @@ RO  ROSA  ROS     ROSAROSAR  ROSARO  RO
 EOF
         ;;
 
+        "rlxos"*)
+            set_colors 12 2 3
+            read -rd '' ascii_data <<'EOF'
+${c1}                                         
+              
+        `"r*?*r, `,;=L\>,`           
+        ?@@@@@@@9Q@@@@@@@Q*          
+        S@@@@@@@@@@@@@@@@@g'         
+       `d@@@@@@@@m^~;D@@@@j`         
+       ,Q@@@@@@@N?`  *hmc,           
+       r@@@@@@@@@@M=`                
+       f@@@@@@@R@@@@y-               
+      `O@@@@@@Q'_D@@@X`              
+      'Q@@@@@@d- :Q@@Q:              
+      'Q@@@@@@u  'B@@@^              
+      'Q@@@@@@O;;E@@@#'              
+       J@@@@@@@@@@@@Q?               
+       `iQ@@@@@@@@Qe:                
+         `:*7v7\;'`                  
+
+EOF
+        ;;
+
         "sabotage"*)
             set_colors 4 7 1
             read -rd '' ascii_data <<'EOF'
-- 
2.32.0

