From 82b5de3049fb87f80fe6c3fde409d4ec31466eff Mon Sep 17 00:00:00 2001
From: itsmanjeet <itsmanjeet1998@gmail.com>
Date: Wed, 10 Jan 2024 18:10:33 +0530
Subject: [PATCH 2/2] workaround overlapping separators when actions items are
 empty

---
 src/Wnck.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/Wnck.cpp b/src/Wnck.cpp
index 1bc054b..c0d106e 100644
--- a/src/Wnck.cpp
+++ b/src/Wnck.cpp
@@ -206,6 +206,7 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 		if (!appInfo->path.empty())
 		{
 			const gchar* const* actions = appInfo->get_actions();
+			bool hasActions = false;
 			for (int i = 0; actions[i]; i++)
 			{
 				// Desktop actions get inserted into the menu above all the window manager controls.
@@ -227,6 +228,8 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 						_appInfo->launch_action((const gchar*)g_object_get_data((GObject*)menuitem, "action"));
 					}),
 					appInfo.get());
+
+				hasActions = true;
 			}
 
 			if (group != nullptr)
@@ -235,7 +238,8 @@ G_GNUC_END_IGNORE_DEPRECATIONS
 				GtkWidget* editLauncher = gtk_menu_item_new_with_label(_("Edit Launcher"));
 
 				gtk_check_menu_item_set_active(GTK_CHECK_MENU_ITEM(pinToggle), group->mPinned);
-				gtk_menu_shell_prepend(GTK_MENU_SHELL(menu), gtk_separator_menu_item_new());
+				if (hasActions)
+					gtk_menu_shell_prepend(GTK_MENU_SHELL(menu), gtk_separator_menu_item_new());
 
 
 				gchar* program = g_find_program_in_path("exo-desktop-item-edit");
-- 
2.42.0

