--- grub-2.06/util/grub.d/10_linux.in	2020-12-23 04:54:17.000000000 +0530
+++ grub-2.06/util/grub.d/10_linux.in.new	2023-09-21 20:15:01.883520956 +0530
@@ -164,7 +164,7 @@
 case "x$machine" in
     xi?86 | xx86_64)
 	list=
-	for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-* ; do
+	for i in /boot/vmlinuz-* /vmlinuz-* /boot/kernel-* /boot/vmlinuz /sysroot/boot/modules/*/bzImage ; do
 	    if grub_file_is_not_garbage "$i" ; then list="$list $i" ; fi
 	done ;;
     *) 
@@ -192,9 +192,15 @@
 submenu_indentation=""
 
 is_top_level=true
+saved_list="${list}"
+for avyos_sys_image in $(swupd sysroot list-images --grub | sort -r) ; do
+    rd_image=$(echo "$avyos_sys_image" | cut -d ':' -f2)
+    avyos_sys_image=$(echo "$avyos_sys_image" | cut -d ':' -f1)
+    gettext_printf "System Image: %s\n" "${avyos_sys_image}" >&2
+    list="${saved_list}"
 while [ "x$list" != "x" ] ; do
   linux=`version_find_latest $list`
-  gettext_printf "Found linux image: %s\n" "$linux" >&2
+  gettext_printf "   linux image: %s\n" "$linux" >&2
   basename=`basename $linux`
   dirname=`dirname $linux`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
@@ -218,7 +222,8 @@
 	   "initramfs-genkernel-${version}" \
 	   "initramfs-genkernel-${alt_version}" \
 	   "initramfs-genkernel-${GENKERNEL_ARCH}-${version}" \
-	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}"; do
+	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}" \
+     "initramfs.img"; do
     if test -e "${dirname}/${i}" ; then
       initrd_real="${i}"
       break
@@ -233,7 +238,7 @@
     for i in ${initrd}; do
       initrd_display="${initrd_display} ${dirname}/${i}"
     done
-    gettext_printf "Found initrd image: %s\n" "$(echo $initrd_display)" >&2
+    gettext_printf "   initrd image: %s\n" "$(echo $initrd_display)" >&2
   fi
 
   config=
@@ -268,35 +273,31 @@
   if [ "x${GRUB_DISABLE_SUBMENU}" = xyes ] || [ "x${GRUB_DISABLE_SUBMENU}" = xy ]; then
     GRUB_DISABLE_SUBMENU="true"
   fi
+  
+  linux_entry "${OS} ${avyos_sys_image}" "${version}" simple \
+  "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} rd.image=${rd_image}"
 
-  if [ "x$is_top_level" = xtrue ] && [ "x${GRUB_DISABLE_SUBMENU}" != xtrue ]; then
-    linux_entry "${OS}" "${version}" simple \
-    "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
-
-    submenu_indentation="$grub_tab"
+  submenu_indentation="$grub_tab"
     
-    if [ -z "$boot_device_id" ]; then
-	boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
-    fi
-    # TRANSLATORS: %s is replaced with an OS name
-    echo "submenu '$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote)' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' {"
-    is_top_level=false
+  if [ -z "$boot_device_id" ]; then
+    boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
   fi
+  # TRANSLATORS: %s is replaced with an OS name
+  echo "submenu '$(gettext_printf "More options for %s" "${OS}" | grub_quote)' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' {"
 
-  linux_entry "${OS}" "${version}" advanced \
-              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+  linux_entry "${OS} ${avyos_sys_image}" "${version}" advanced \
+              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} rd.image=${rd_image}"
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
-    linux_entry "${OS}" "${version}" recovery \
-                "single ${GRUB_CMDLINE_LINUX}"
+    linux_entry "${OS} ${avyos_sys_image}" "${version}" recovery \
+                "single ${GRUB_CMDLINE_LINUX} rd.image=${rd_image}"
   fi
+  
+  echo '}'
+ 
 
   list=`echo $list | tr ' ' '\n' | fgrep -vx "$linux" | tr '\n' ' '`
 done
+done
 
-# If at least one kernel was found, then we need to
-# add a closing '}' for the submenu command.
-if [ x"$is_top_level" != xtrue ]; then
-  echo '}'
-fi
 
 echo "$title_correction_code"
