--- meson.build	2023-10-24 13:09:24.000000000 +0300
+++ meson.build	2023-10-28 11:31:00.628950048 +0300
@@ -13,6 +13,7 @@
 
 add_project_arguments('-Wno-unused-parameter', language: 'c')
 add_project_arguments('-Wno-missing-braces', language: 'c')
+add_project_arguments('-Wno-deprecated-declarations', language: 'c')
 
 prefix = get_option('prefix')
 localedir = join_paths(prefix, get_option('localedir'))
